superset/assets/src/CsvToDatabase/CsvToDatabase.jsx

```diff
@@ -0,0 +1,497 @@
+    this.setTableExists = this.setTableExists.bind(this);
+    this.setUserInput = this.setUserInput.bind(this);
+    this.getConnectionStrings = this.getConnectionStrings.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  setFile(file) {
+    if (file && file[0]) {
+      let fileName = '';
+      if (this.state.selectedConnection.value === -1) {
+        fileName = file[0].name.slice(0, -4);
+      }
+      this.setState({ file: file[0], databaseName: fileName });
+    }
+  }
+
+  setSelectedConnection(connection) {
+    let databaseName = '';
+    if (connection.value === -1 && this.state.file) {
+      databaseName = this.state.file.name.slice(0, -4);
+    }
+    this.setState({
+      selectedConnection: connection,
+      databaseName,
+    });
+  }
+
+  setTableExists(value) {
+    this.setState({ selectedTableExists: value });
+  }
+
+  setUserInput(event) {
+    const name = event.target.name;
+    const value = event.target.value;
+    this.setState({ [name]: value });
+  }
+
+  setCheckboxValue(name, value) {
+    this.setState({ [name]: value });
+  }
+
+  getConnectionStrings() {
+    const connections = [];
+    this.props.databases.forEach(database =>
+      connections.push({ label: database.name, value: database.id }),
+    );
+    return connections;
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const {
+      tableName,
+      databaseName,
+      file,
+      selectedConnection,
+      schema,
+      delimiter,
+      selectedTableExists,
+      headerRow,
+      decimalCharacter,
+      indexColumn,
+      mangleDuplicateColumns,
+      skipInitialSpace,
+      skipRows,
+      rowsToRead,
+      skipBlankLines,
+      parseDates,
+      inferDatetimeFormat,
+      dataframeIndex,
+      columnLabels,
+    } = this.state;
+    this.props.actions.uploadCsv({
+      tableName,
+      databaseName,
+      file,
+      connectionId: selectedConnection.value,
+      schema,
+      delimiter,
+      ifTableExists: selectedTableExists.value,
+      headerRow,
+      decimalCharacter,
+      indexColumn,
+      mangleDuplicateColumns,
+      skipInitialSpace,
+      skipRows,
+      rowsToRead,
+      skipBlankLines,
+      parseDates,
+      inferDatetimeFormat,
+      dataframeIndex,
+      columnLabels,
+    });
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <FormError status={this.props.uploadStatus} />
+        <div className="panel panel-primary">
+          <div className="panel-heading">
+            <h4 className="panel-title">{t('CSV to Database configuration')}</h4>
+          </div>
+          <div id="Home" className="tab-pane active">
+            <form
+              id="model_form"
+              method="post"
+              encType="multipart/form-data"
+              onSubmit={this.handleSubmit}
+            >
+              <div className="table-responsive">
+                <table className="table table-bordered">
```

Comment:
For now, since we still have some issues regarding the backend which we want to prioritize, we will leave the UI as is. Changing it in the future should not be a big issue but for this PR we will go with what we implemented now if that's OK.